### PR TITLE
Created type file for adlib repo

### PR DESCRIPTION
**Pull request by MikeTschudi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/adlib/adlib-tests.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+import * as adlib from "adlib";
+
+const transform1: adlib.TransformFunction =
+    (key: string, value: any, settings: any, param?: any): any => {
+        return null;
+    };
+
+    const transformsList: adlib.TransformsList = {
```
Comment (DanielRosenwasser) [inline types/adlib/adlib-tests.ts]: Nit: indentation is a little off

File: types/adlib/adlib-tests.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    firstXform: transform1
+};
+
+const template: any = {
```
Comment (DanielRosenwasser) [inline types/adlib/adlib-tests.ts]: Any reason you need the `any`s?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/adlib/adlib-tests.ts (hunk 1 above)
Comment (MikeTschudi) [reply types/adlib/adlib-tests.ts]: Thanks for the catch; fixed

File: types/adlib/adlib-tests.ts (hunk 2 above)
Comment (MikeTschudi) [reply types/adlib/adlib-tests.ts]: Nope--just ignorance on my part!  Removed them.

**Comment by MikeTschudi** [comment]
@DanielRosenwasser, does this PR have the right labels?  The suggested improvements have been made and are ready for a 2nd review. Thanks

**Comment by DanielRosenwasser** [comment]
Thanks for the patience!

**Comment by MikeTschudi** [comment]
Thanks, Daniel!  I just wasn't sure if the bot accidentally had the PR in the "awaiting author revisions" category.

**Pull request merged**